src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const nextState = { ...currentState };
+
```

Comment:
Instead of creating 2 copies of state, let's keep one and when you are pushing it to the history use spread operator stateHistory.push({...currentState})